L1Trigger/TrackTrigger/plugins/TTStubBuilder.cc

```diff
@@ -0,0 +1,327 @@
+          }
+          std::sort( bendMap.begin(), bendMap.end(), TTStubBuilder< Ref_Phase2TrackerDigi_ >::SortStubBendPairs );
+
+          for ( unsigned int i = 0; i < maxStubs; ++i )
+          {
+            /// Put the highest momenta (lowest bend) stubs into the event
+            tempInner.push_back( *(is.second[bendMap[i].first].getClusterRef(0)) );
+            tempOuter.push_back( *(is.second[bendMap[i].first].getClusterRef(1)) );
+            tempAccepted.push_back( is.second[bendMap[i].first] );
+          }
+        }
+      } /// End of loop over temp output
+    } /// End store only the selected stubs if max no. stub/ROC is set
+    /// Create the FastFillers
+    if ( tempInner.size() > 0 )
+    {
+      typename edmNew::DetSetVector< TTCluster< Ref_Phase2TrackerDigi_ > >::FastFiller lowerOutputFiller( *ttClusterDSVForOutput, lowerDetid );
+      for ( unsigned int m = 0; m < tempInner.size(); m++ )
+      {
+        lowerOutputFiller.push_back( tempInner.at(m) );
+      }
+      if ( lowerOutputFiller.empty() )
+        lowerOutputFiller.abort();
+    }
+
+    if ( tempOuter.size() > 0 )
+    {
+      typename edmNew::DetSetVector< TTCluster< Ref_Phase2TrackerDigi_ > >::FastFiller upperOutputFiller( *ttClusterDSVForOutput, upperDetid );
+      for ( unsigned int m = 0; m < tempOuter.size(); m++ )
+      {
+        upperOutputFiller.push_back( tempOuter.at(m) );
+      }
+      if ( upperOutputFiller.empty() )
+        upperOutputFiller.abort();
+    }
+
+    if ( tempAccepted.size() > 0 )
+    {
+      typename edmNew::DetSetVector< TTStub< Ref_Phase2TrackerDigi_ > >::FastFiller tempAcceptedFiller( *ttStubDSVForOutputTemp, stackDetid);
+      for ( unsigned int m = 0; m < tempAccepted.size(); m++ )
+      {
+        tempAcceptedFiller.push_back( tempAccepted.at(m) );
+      }
+      if ( tempAcceptedFiller.empty() )
+        tempAcceptedFiller.abort();
+    }
+
+  } /// End of loop over detector elements
+
+  /// Put output in the event (1)
+  /// Get also the OrphanHandle of the accepted clusters
+  edm::OrphanHandle< edmNew::DetSetVector< TTCluster< Ref_Phase2TrackerDigi_ > > > ttClusterAcceptedHandle = iEvent.put( std::move(ttClusterDSVForOutput), "ClusterAccepted" );
+
+  /// Now, correctly reset the output
+  typename edmNew::DetSetVector< TTStub< Ref_Phase2TrackerDigi_ > >::const_iterator stubDetIter;
+
+  for ( stubDetIter = ttStubDSVForOutputTemp->begin();
+        stubDetIter != ttStubDSVForOutputTemp->end();
+        ++stubDetIter ) {
+    /// Get the DetId and prepare the FastFiller
+    DetId thisStackedDetId = stubDetIter->id();
+    typename edmNew::DetSetVector< TTStub< Ref_Phase2TrackerDigi_ > >::FastFiller acceptedOutputFiller( *ttStubDSVForOutputAccepted, thisStackedDetId );
+
+    /// detid of the two components. 
+    ///This should be done via a TrackerTopology method that is not yet available.
+    DetId lowerDetid = thisStackedDetId+1;
```

Comment:
please implement a helper function in TrakerTopology with this code - it will be easier to generalize if its there.